components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java

```diff
@@ -491,7 +494,10 @@ private JSONArray joinArrays(JSONArray jsonOne, JSONArray jsonTwo)
     private void updateDisordersFromJSON(XWikiDocument doc, BaseObject data, XWikiContext context, JSONObject json)
     {
         try {
-            JSONArray inputDisorders = json.optJSONArray(JSON_KEY_DISORDERS);
+            if (!json.has(JSON_KEY_DISORDERS)) {
+                return;
```

Comment:
why this change? optJSONArray(JSON_KEY_DISORDERS) would return null if either there is no key or key value is not an array (e.g. misformatted JSON)

with new code, no key is covered, by value not being a proper JSON array will result in a JSONException